subsys/net/lib/nrf_cloud/src/nrf_cloud_log_backend.c

```diff
@@ -132,8 +132,8 @@ BUILD_ASSERT(!((IS_ENABLED(CONFIG_LOG_BACKEND_UART_OUTPUT_TEXT) ||
 		"CONFIG_LOG_FMT_SECTION_STRIP is not compatible with text logging.");
 
 #if defined(CONFIG_NRF_CLOUD_LOG_DICTIONARY_LOGGING_ENABLED)
-BUILD_ASSERT(IS_ENABLED(CONFIG_NRF_CLOUD_MQTT),
-	     "nRF Cloud dictionary logging is only available for MQTT");
+BUILD_ASSERT(!IS_ENABLED(CONFIG_NRF_CLOUD_REST),
+	     "nRF Cloud dictionary logging is only available for MQTT and CoAP");
```

Comment:
Personally I'd prefer to see a check for `CLOUD_COAP || CLOUD_MQTT` -- if we ever add a fourth transport, this could be a landmine waiting for us

Maybe a computed Kconfig value or #define instead of direct check

Reply:
It doesn't matter because REST support will merge first, in which case the combination of this PR on top of main will eliminate this section after rebase and conflict resolution.